unittests/test_fields.py

```diff
@@ -82,31 +84,25 @@ class FieldTester:
                 'field_maybe_none', type(None))
 
         tester = FieldTester()
-        tester.field = (65, 22, 47)
+        tester.field = '1d65h22m87s'
         tester.field_maybe_none = None
 
-        self.assertIsInstance(FieldTester.field, fields.TimerField)
-        self.assertEqual((65, 22, 47), tester.field)
-        self.assertRaises(TypeError, exec, 'tester.field = (2,)',
-                          globals(), locals())
-        self.assertRaises(TypeError, exec, 'tester.field = (2, 2)',
-                          globals(), locals())
-        self.assertRaises(TypeError, exec, 'tester.field = (2, 2, 3.4)',
-                          globals(), locals())
-        self.assertRaises(TypeError, exec, "tester.field = ('foo', 2, 3)",
-                          globals(), locals())
-        self.assertRaises(TypeError, exec, 'tester.field = 3',
-                          globals(), locals())
-        self.assertRaises(ValueError, exec, 'tester.field = (-2, 3, 5)',
-                          globals(), locals())
-        self.assertRaises(ValueError, exec, 'tester.field = (100, -3, 4)',
-                          globals(), locals())
-        self.assertRaises(ValueError, exec, 'tester.field = (100, 3, -4)',
-                          globals(), locals())
-        self.assertRaises(ValueError, exec, 'tester.field = (100, 65, 4)',
-                          globals(), locals())
-        self.assertRaises(ValueError, exec, 'tester.field = (100, 3, 65)',
-                          globals(), locals())
+        assert isinstance(FieldTester.field, fields.TimerField)
+        assert (datetime.timedelta(days=1, hours=65, minutes=22,
+                                   seconds=87) == tester.field)
+        tester.field = ''
+        assert (datetime.timedelta(days=0, hours=0, minutes=0,
+                                   seconds=0) == tester.field)
+        with pytest.raises(ValueError):
+            exec('tester.field = "1e"', globals(), locals())
+            exec('tester.field = "-10m5s"', globals(), locals())
+            exec('tester.field = "10m-5s"', globals(), locals())
+            exec('tester.field = "m10s"', globals(), locals())
+            exec('tester.field = "10m10"', globals(), locals())
+            exec('tester.field = "10m10m1s"', globals(), locals())
+            exec('tester.field = "10m5s3m"', globals(), locals())
+            exec('tester.field = "10ms"', globals(), locals())
+            exec('tester.field = 10', globals(), locals())
```

Comment:
You should also add the following cases:

1. Test the case when you assign directly a `timedelta` object.
2. Test that a deprecation warning is raised when you do the old-style assignment.